lib/spack/spack/cmd/common/arguments.py

```diff
@@ -185,7 +185,8 @@ def installed_specs():
 def yes_to_all():
     return Args(
         '-y', '--yes-to-all', action='store_true', dest='yes_to_all',
-        help='assume "yes" is the answer to every confirmation request')
+        help="""assume "yes" is the answer to every confirmation request
+to run completely non-interactively, also specify--no-checksum.""")
```

Comment:
I looked at this again and noticed that this particular help string only makes sense for `spack install` -- it's the only command that has *both* `no_checksum` and `yes_to_all`:

```console
$ cd lib/spack/spack/cmd
$ git grep -l no_checksum | sort > a  # a is a sorted list of filenames that have no_checksum
$ git grep -l yes_to_all | sort > b   # b is a sorted list of filenames that have yes_to_all
$ comm -12 a b                        # find common lines in sorted files
common/arguments.py
install.py
```

With this change, *all* places that have a `yes_to_all` option are affected, so all these commands:

```console
$ git grep -l yes_to_all
common/arguments.py
config.py
deprecate.py
env.py
gc.py
install.py
modules/__init__.py
uninstall.py
```

I think what we should do here is not modify this in `common/arguments.py`, but special-case the argument for spack install.  So:

1. add another `subparser.add_argument()` call in `cmd/install.py` with the arguments above
2. remove `'yes_to_all'` from the call to `arguments.add_common_arguments(subparser, ['yes_to_all', 'spec'])` in `cmd/install.py`
3. leave `common/arguments.py` alone after all (i.e., undo the change above)